svgscreen/svgscreen.go

```diff
@@ -78,18 +127,61 @@ func Render(w io.Writer, s Screen) error {
 				}
 				l.Chars[i] = c
 			}
+		}
+	}
 
+	// Render the whole background first. Then it will not be selected when copy/pasting from rendered SVG
+	for _, l := range s.Lines {
+		bg := LineToTextElement(s, l, func(c Char) TextSpan {
 			if c.Background == "" {
-				continue
-			}
-			if strings.HasPrefix(c.Background, "#") {
-				backgroundColorsUsed[c.Background] = c.Background
+				return TextSpan{
+					Style:   "",
+					Content: " ",
+				}
 			} else {
-				backgroundColorsUsed[c.Background] = s.BackgroundColors[c.Background]
+				return TextSpan{
+					Style: template.CSS(
+						"fill:" + ResolveColor(c.Background, s.BackgroundColors) + "; text-shadow: -0.01px -0.01px, 0.01px -0.01px, -0.01px 0.01px, 0.01px 0.01px;"),
+					Content: "█",
+				}
+			}
+		})
+		if len(bg.TextSpans) > 0 {
+			s.TextElements = append(s.TextElements, bg)
+		}
+	}
+
+	// Then render the foreground
+	for _, l := range s.Lines {
+		fg := LineToTextElement(s, l, func(c Char) TextSpan {
+			var styles []string
+			deco := ""
+
+			if c.Foreground != "" {
+				styles = append(styles, "fill:"+ResolveColor(c.Foreground, s.ForegroundColors))
+			}
+			if c.Intensity {
+				styles = append(styles, "font-weight:bold")
```

Comment:
This might explain why the `jq.ansi` test looks a bit darker now. Before it did this:
```
        {{- if $c.Intensity}}
            {{- $color = coloradd $color "#505050"}}
        {{- end}}
```
This is how iterm2 renders the `b-it-ul-st.ansi` test:
![Screenshot 2023-12-21 at 12 23 58](https://github.com/wader/ansisvg/assets/185566/5d47c109-9eeb-49bf-bc1e-9f0d284aeeab)

So maybe we should do both bold and increase brightness somewhat?

Reply:
Ok. Maybe it makes sense to have this amount variable per color scheme. (if there will be color schemes some time ...)

Reply:
Took a deeper look. For standard ANSI palette non-bright colors it seems to just pick the "bright" variant, for others i'm not sure. In iterm2 it seems the color changes a tiny bit but maybe it's just the font boldness making it look a bit brighter?

```
#!/bin/sh
#!/bin/sh
for c in '' "\e[31m" "\e[90m" "\e[91m" "\e[38;2;100;0;0;40m"; do
printf "${c}normal\e[0m \e[1m${c}intensive\e[0m\n"
done
```
renders as:
![Screenshot 2023-12-21 at 16 16 17](https://github.com/wader/ansisvg/assets/185566/88103ec7-851e-4ed1-b00e-bf176e6383a2)

poked around in https://github.com/mbadolato/iTerm2-Color-Schemes (where the ansisvg colors schemes comes from) and can't see that there is anything specified per scheme except the palette bright colors

Reply:
this is how it looks in kitty, the colors seem to be the same left and right per row

![image](https://github.com/wader/ansisvg/assets/17723608/510066f6-b2b2-4d3c-862b-0912993d5a20)

Reply:
same in gnome terminal
![image](https://github.com/wader/ansisvg/assets/17723608/106a2ba5-507e-4068-a403-baf5ed3501ef)

Reply:
Ok thanks for checking. That makes things a bit easier and also he current behaviour is not correct? is always adds brightness for intensity but should only switch to bright variant for normal palette color if i understand it correctly